src/execution_plan/execution_plan.c

```diff
@@ -290,11 +290,11 @@ static void _ExecutionPlan_ProcessQueryGraph(ExecutionPlan *plan, QueryGraph *qg
 				if(AlgebraicExpression_OperandCount(exp) == 0) continue;
 
 				QGEdge *edge = NULL;
-				const char *alias = AlgebraicExpression_Edge(exp);
-				if(AlgebraicExpression_Edge(exp)) edge = QueryGraph_GetEdgeByAlias(qg, alias);
+				const char *edge_alias = AlgebraicExpression_Edge(exp);
+				if(edge_alias) edge = QueryGraph_GetEdgeByAlias(qg, edge_alias);
 				if(edge && (QGEdge_VariableLength(edge) ||
 							(QGEdge_MultipleHops(edge) &&
-							 AST_AliasReferencReasons(ast, alias, REFERENCED_ENTITY_PART_OF_NAMED_PATH)))) {
+							 (REFERENCED_ENTITY_PART_OF_NAMED_PATH & AST_AliasReferencReasons(ast, edge_alias))))) {
```

Comment:
I think it would a bit cleaner to dedicate a boolean variable for each function and inspect those within the condition,
e.g.
```
bool var_len = QGEdge_VariableLength(edge);
bool multi_hop = QGEdge_MultipleHops(edge);
...
```